onnxruntime/core/providers/coreml/builders/model_builder.cc

```diff
@@ -71,9 +71,47 @@ void CopyRawDataToRepeatedField(const ONNX_NAMESPACE::TensorProto& tensor_proto,
   }
 }
 
+template <>
+void CopyRawDataToRepeatedField<int64_t, int32_t>(const ONNX_NAMESPACE::TensorProto& tensor_proto,
+                                                  google::protobuf::RepeatedField<int32_t>& repeated_field) {
+  const auto& raw_data = tensor_proto.raw_data();
+  const int64_t* data = reinterpret_cast<const int64_t*>(raw_data.data());
+  const size_t element_count = raw_data.size() / sizeof(int64_t);
+  const std::string tensor_name = tensor_proto.name();
+
+  // Reserve space to avoid multiple reallocations
+  repeated_field.Reserve(narrow<int>(element_count));
+
+  // Use std::transform with proper iterators
+  std::transform(data, data + element_count,
+                 google::protobuf::RepeatedFieldBackInserter(&repeated_field),
+                 [&tensor_name](int64_t v) {
+                   try {
+                     return narrow<int32_t>(v);
+                   } catch (const std::exception& e) {
+                     ORT_THROW("Error converting int64 to int32 for tensor ", tensor_name, ": ", e.what(),
+                               ". Value (", v, ") exceeds int32 range.");
+                   }
```

Comment:
it's nice that you're including more detail in the message.

however, generally, we try to avoid using `try` or `catch` directly in order to support builds with exceptions disabled. that's why those `ORT_THROW` and `ORT_CATCH` macros are there.

in this case, I think we can just call `narrow` and not bother with adding more info to the exception. if exceptions are enabled, it should be possible to follow a call stack back to this code.